dali/driver/base.py

```diff
@@ -1,5 +1,7 @@
 import threading
 import usb
+import serial
```

Comment:
This import and the usb import should be conditional.

Reply:
I would therefore add a ``try ... except`` around the import. Do you have any suggestions on what to do when the import fails? Do some logging? Raise a warning?

Reply:
My favourite pattern would be an inline import

    class Foo:
        def __init__(self):
            try:
                import serial
            except ImportError:
                raise RuntimeError('Foo requires pyserial but it is not installed')
            self.conn = serial.Serial()

Reply:
as alternative you can use a marker

    try:
        import serial
        SERIAL_INSTALLED = True
    except ImportError:
        SERIAL_INSTALLED = False

    class Foo:
        def __init__(self):
            if not SERIAL_INSTALLED:
                raise RuntimeError('Foo requires pyserial but it is not installed')